### PR TITLE
fix(listitem): get stories to render and snapshot, fix proptype warnings

### LINKED ISSUE #1410: [List] propType warning
### Describe the bug

The following warning is given when using HierarchyList, but I believe its coming from the underlying ListItem

Screen Shot 2020-07-17 at 12 11 44 PM

### To Reproduce

1. Go to implementation of a List
2. See propType warning
- It is intermittent as I think it requires just the right props to throw the warning for some reason

### Environment/versions:

- OS:  MacOS
- Browser:  chrome
- `carbon-addons-iot-react` version: 2.100.0

### Specific timeline issues / requests

No specific time

### LINKED ISSUE #1412: [SimpleList / ListItem / TileCatalogNew] fix snapshot unique key warnings
### Describe the bug

Multiple unique key warnings are being thrown in SimpleList, ListItem, and TileCatalogNew. I've added a snapshot exception  in https://github.com/IBM/carbon-addons-iot-react/pull/1414 for all 3 components as they're causing the snapshots to fail 

### To Reproduce

1. Re-enable the snapshots by removing the regex exception in `StorybookSnapshots.test.js`
2. Run `yarn test:update`
3. See unique key warnings

### Expected behavior

Snapshots should pass with no warnings or errors

### Environment/versions:

- OS: MacOS
- Browser: chrome
- `carbon-addons-iot-react` version: v2.100.2

### Specific timeline issues / requests

Do you want this work within a specific time period? Is it related to an
upcoming release?

No specific timeline, but non-unique keys can cause unexpected behavior in production

### LINKED ISSUE #1664: [ListItem] stories are not rendering
### What package is this for?

- [x] React
- [ ] Angular

### Describe the bug

Stories are not rendering for the ListItem
https://carbon-addons-iot-react.com/?path=/story/watson-iot-experimental-listitem--basic-w-knobs

### To Reproduce

1. Go to https://carbon-addons-iot-react.com/?path=/story/watson-iot-experimental-listitem--basic-w-knobs
2. See that this and all other ListItem stories are not rendering. There is no error in the console

### Environment/versions:

- OS: MacOS
- Browser: chrome
- `carbon-addons-iot-react` version: `2.129.1`

### Additional context

Potentially could be solved via #963

### PR DESCRIPTION
**Pull request by tay1orjones**
Closes #1664
Closes #1412 
Closes #1410 

**Summary**
- Updated the ListItem stories so they render and fixed some related proptype/key warnings preventing snapshot testing of these stories. 

**Change List (commits, features, bugs, etc)**

- Update ListItem stories to render the root unconnected component, provide stub react-dnd props
- Fix proptype and "unique `key`" warnings for SimpleList, ListItem, TileCatalogNew
- Remove the snapshot exception for those 3 components

**Acceptance Test (how to verify the PR)**

- Validate that [the ListItem stories](https://deploy-preview-1669--carbon-addons-iot-react.netlify.app/) are working
- Validate no local console proptype/key warnings for all SimpleList, ListItem, TileCatalogNew stories

### CODE REVIEW

File: src/components/List/ListItem/ListItem.story.jsx [hunk 1]
```diff
@@ -62,6 +76,8 @@ storiesOf('Watson IoT Experimental/ListItem', module)
     return (
       <div style={{ width: 400 }}>
         <ListItem
+          {...dndProps}
+          {...dndProps}
```
Comment (sstone2423) [inline src/components/List/ListItem/ListItem.story.jsx:80]: Looks like 1 extra prop pass is here

**Review by sstone2423 — APPROVED** [review]
Thanks for this!

File: src/components/List/ListItem/ListItem.story.jsx (hunk 1 above)
Comment (tay1orjones) [reply src/components/List/ListItem/ListItem.story.jsx]: 🤦 fixed, thanks!

File: src/components/Dashboard/CardRenderer.jsx [hunk 2]
```diff
@@ -61,148 +61,161 @@ export const loadCardData = async (card, setCard, onFetchData, timeGrain) => {
...
-  onCardAction, i18n, isLoading, isEditable, breakpoint, onFetchData, onSetupCard, renderIconByName, timeGrain, ...cardProp }) => {
-  // pass through the card props
-  /* eslint-enable react/prop-types */
-  if (__DEV__) {
-    warning(
-      false,
-      'CardRenderer component has been deprecated and will be removed in the next release of `carbon-addons-iot-react`.'
-    );
-  }
+const CardRenderer = React.memo(
```
Comment (bjornalm) [inline src/components/Dashboard/CardRenderer.jsx:64]: Is this whole component just reformatted and no real changes?

**Review by bjornalm — APPROVED** [review]
Nice work

**Pull request merged**